en/nearest-neighbor-search.md

```diff
@@ -5,36 +5,82 @@ redirect_from:
 - /documentation/nearest-neighbor-search.html
 ---
 
-Searching for the nearest neighbors of a data point in high dimensional vector space is a 
-fundamental problem for many real-world applications.
+Nearest neighbor search, or vector search, is a technique used
+to find the closest data points to a given query point in a high-dimensional vector space.
+This is supported in Vespa using the
+[nearestNeighbor](reference/query-language-reference.html#nearestneighbor) query operator.
+This operator can be combined with other filters or query terms using the
+[Vespa query language](query-language.html),
+making it easy to create hybrid solutions that combine modern vector based techniques with
+[traditional information retrieval](text-matching-ranking.html).
 
-Many real-world applications require query-time constrained vector search. 
-For example, real-time recommender systems using [vector search for candidate retrieval](tutorials/news-5-recommendation.html) 
-need to filter recommendations by hard constraints like regional availability or age group suitability. 
-Likewise, search systems using vector search need to support filtering.
-For example, typical e-commerce search interfaces allow users to navigate and filter search results using result facets. 
 
-## Using Vespa's nearest neighbor search
+## Vectors in Vespa
+In Vespa a vector is represented by a [tensor](tensor-user-guide.html) with one indexed dimension.
+Example [tensor type](reference/tensor.html#tensor-type-spec) representing a float vector with 384 dimensions:
```

Comment:
Maybe we should link to the cell precision types supported by tensors and nearest neighbor search

Reply:
The linked doc (reference/tensor.html#tensor-type-spec) lists the four cell precision types: double, float, bfloat16, int8, so this should be ok.